src/libstore/gc.cc

```diff
@@ -330,18 +335,184 @@ Roots LocalStore::findRoots()
 }
 
 
+static void readProcLink(int procFd, const char * file, StringSet & paths)
```

Comment:
@edolstra I can't use `readLink` here because `lstat` on files in proc always returns 0 for size.